src/backend/distributed/README.md

```diff
@@ -1744,9 +1744,141 @@ The reason for handling dependencies and deparsing in post-process step is that
+ ...
+
+-- Then, check pg_dist_object. This should be consistent across all nodes.
+SELECT
+    pg_identify_object_as_address(classid, objid, objsubid) as distributed_object
+FROM
+    pg_dist_object
+WHERE
+    (classid, objid, objsubid) 
+        IN 
+    (SELECT classid, objid, objsubid FROM pg_get_object_address('type', '{type_test}', '{}'));
+┌──────────────────────────────┐
+│      distributed_object      │
+├──────────────────────────────┤
+│ (type,{public.type_test},{}) │
+└──────────────────────────────┘
+(1 row)
+```
+
+In rare cases, `pg_dist_object` is updated during Citus version upgrades. If you upgrade Citus from `version X` to `version Y`, and a certain object becomes a supported distributed object in `version Y` but wasn't in `version X`, Citus marks it as such during the `ALTER EXTENSION citus` command. The details can be found in the C function `PostprocessAlterExtensionCitusUpdateStmt()`.
+
+### How Are `pg_dist_object` and `pg_depend` Related?
+
+In the prior sections, we focused on standalone objects, but in reality, most objects have dependencies. That's where `pg_depend` and also `pg_shdepend` become crucial. Any mention of `pg_depend` in this section also applies to `pg_shdepend`.
+
+When Citus creates an object, it scans the `pg_depend` table to identify all dependencies and then automatically generates these dependencies as distributed objects. 
+
+The core C function in this process is `EnsureDependenciesExistOnAllNodes()`. This function takes the object as an argument and conducts a depth-first search (DFS) on `pg_depend` and `pg_shdepend` tables. The DFS sequence is crucial because dependencies must be established in a specific order. For instance, if a table depends on a type `t1`, and `t1` relies on another type `t2`, then `t2` must be created before `t1`. The DFS ensures that this order is maintained.
+
+If Citus encounters a dependency it can't support, it will throw an error instead of silently allowing it. The rationale behind this approach is to avoid subtle issues later, especially when adding new nodes. For instance, Citus currently throws an error for circular dependencies. The main function performing these checks is `EnsureDependenciesCanBeDistributed()`.
+
+During the DFS, Citus might need to extend its search, especially for complex dependencies like array types that don't have a straightforward dependency on their element types. Citus expands the traversal to account for such cases. The main function responsible for this is `ExpandCitusSupportedTypes()`, which has extensive comments explaining the specific rules.
+
+### `current_user` vs `superuser` for Object Propagation:
```

Comment:
I was a bit confused by the use of `current_user` here, since that is syntax specific to things like `ALTER ROLE current_user`. Maybe just current user